Updated the flow control tutorial (#1287)
ARMmbed/Handbook: docs/api/io/flow_control.md

If we want to automatically blink an LED, we have three main techniques:

1. [Wait](#wait)
1. [Ticker](#ticker)
1. [Thread](#thread)

### Wait

#### Busy wait

Busy wait is a method that blocks the processor for a period of time. This is an effective way to create time delays, but it is inefficient because it wastes processor time and keeps the processor running at full power for the duration of the wait:

[![View code](https://www.mbed.com/embed/?url=https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/Tutorials_UsingAPIs/Flow-Control-Busy-Wait)](https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/Tutorials_UsingAPIs/Flow-Control-Busy-Wait/main.cpp)

Notice `printf()`; you can enable this by uncommenting the line (remove the `//`). `printf()` prints to the terminal, so you can use it to get debug information.

> <span class="tips">**Tip:** Busy wait is inefficient and should be avoided. </span>

Reviewer: ```suggestion
<span class="tips">**Tip:** Busy wait is inefficient, and we recommend not using it. </span>
```

#### Suspend thread execution

> A better alternative to busy wait is to suspend thread execution while waiting. You can use the `ThisThread::sleep_for` C++ API or `thread_sleep_for` C API.

Reviewer: ```suggestion
A better alternative to busy wait is to suspend thread execution while waiting. Use:

 * For C++: The `ThisThread::sleep_for` API 
 * For C: The `thread_sleep_for` API.
```

### Ticker

Tickers and timers are another way of creating a time interval. These methods allow other code to run while you are waiting. During the wait period, if no threads are running, the idle thread can automatically put the system to sleep.

> If you don't need the precision of a high-frequency timer/ticker, we recommend that you use LowPowerTimer/LowPowerTicker instead because this will allow the system to be put in deep sleep mode.

Reviewer: ```suggestion
If you don't need the precision of a high-frequency timer or ticker, we recommend that you use LowPowerTimer or LowPowerTicker instead. This allows the system to be put in deep sleep mode.
```
